### PR TITLE
ARXIVNG-299 added extra fields to advanced and simple search

### PR DESCRIPTION
**Pull request by erickpeirson**
image

Also updated help text on the advanced search (@eawoods please veto/override)

image

Here's how they're handled:

- title, abstract, comments: supports phrase-based search with quotes, executed as a ``simple_query_string``
- authors: match (as before); includes author-owners
- acm_class, msc_class, doi: keyword match
- paper_id: keyword match with or without version 
- journal_ref, report_num: ``match_phrase`` -- this will give a precise match for specific searches (e.g. full jref), but also support broader searches (e.g. by journal name).

I didn't touch categories at this point, but do we want to include that in the simple search or field terms? I know we talked about it, but found myself to be unclear on precisely what we decided.

### CODE REVIEW

File: search/services/index.py [hunk 1]
```diff
@@ -159,27 +159,45 @@ def _group_terms(query: Query) -> tuple:
         return terms[0]
 
     @staticmethod
-    def _field_term_to_q(term) -> Q:
-        if term.field in ['title', 'abstract']:
+    def _field_term_to_q(field: str, term: str) -> Q:
```
Comment (erickpeirson) [inline search/services/index.py]: This is now used by both advanced and simple search, since we're doing precisely the same thing with each field.

**Comment by erickpeirson** [comment]
Included report number to support #75

**Comment by jimentwood** [comment]
This is looking good to me It may be helpful to get more specific user input about how search is used after beta is launched and people kick the tires.
Is this picking up on submitter name at all? I'm not sure how useful that is, but the current search includes submitter.

**Comment by erickpeirson** [comment]
We haven't added submitter explicitly. Author searches are based on both the parsed author names (from the canonical author name string) and on the author-owners; @mhl10 should correct me if I'm getting this wrong, but I believe that this should include the submitter unless it was a proxy submission.

That being said, it would be pretty easy to add submitter as an explicit search option.

**Comment by mhl10** [comment]
@erickpeirson it should be OK to include submitter whenever the `is_author` field is set to `true` and it's not a proxy submission. 

In most cases the submitter name will be exactly the same as one name in `owners` but since the submitter name is recorded at submission creation time it could be different, because `owners` is derived from data in the user tables.

**Comment by erickpeirson** [comment]
Ok; so if I'm understanding correctly: 
- Categories in field-based search: hold off
- Add support to search by submitter

**Comment by eawoods** [comment]
I think so. @mhl10 ?

**Comment by mhl10** [comment]
My previous comment was a bit vague. I think we should include the submitter in the *author* search on the condition the `is_author` flag is true.  I think it's OK to have a separate advanced search field for submitter too, but am wondering if that would be a bit confusing for some users; I think some don't realize that submitters aren't necessarily authors.

**Comment by erickpeirson** [comment]
Got it. Makes sense. Will implement those two changes and then merge.

**Comment by erickpeirson** [comment]
image

**Comment by erickpeirson** [comment]
Fixes #68

**Pull request merged**